examples/client_replication/src/client.rs

```diff
@@ -52,23 +52,26 @@ pub(crate) fn init(mut client: ResMut<ClientConnection>) {
 pub(crate) fn handle_connection(
     mut commands: Commands,
     mut connection_event: EventReader<ConnectEvent>,
+    config: Res<ClientConfig>,
 ) {
     for event in connection_event.read() {
         let client_id = event.client_id();
-        commands.spawn(TextBundle::from_section(
-            format!("Client {}", client_id),
-            TextStyle {
-                font_size: 30.0,
-                color: Color::WHITE,
-                ..default()
-            },
-        ));
-        // spawn a local cursor which will be replicated to other clients, but remain client-authoritative.
-        commands.spawn(CursorBundle::new(
-            client_id,
-            Vec2::ZERO,
-            color_from_id(client_id),
-        ));
+        if config.shared.mode == Mode::Separate || client_id.is_local() {
```

Comment:
Which cases are excluded from this condition? My brain is having trouble parsing this.

Reply:
Yes I think this is actually not needed: the issue was that there is some code handling client-side ConnectEvents and some code handling server-side ConnectEvents.

In this PR I had merged the 2 into a single one, so even when another client than the local client was connecting, it was also triggering the client-side ConnectEvent code. I will separate the two events and this change probably won't be needed anymore